default/python/AI/AIDependencies.py

```diff
@@ -49,6 +49,15 @@
 TECH_COST_MULTIPLIER = 2.0
 FOCUS_CHANGE_PENALTY = 1
 STABILITY_PER_LIKED_FOCUS = 2.0
+HOMEWORLD_BONUS = 5.0
+STABILITY_PER_LIKED_BUILDING_ON_PLANET = 4.0
```

Comment:
Can values like these be queried from the scripting files where they are defined?
I'm thinking of ease of maintenance. Modders (like me) changing the named values in the scripting files might completely miss the corresponding AI files.

Reply:
Let's bring validation of the constants used by AI to the next voice meeting.

Reply:
Since April the AI can read NamedInt and NamedReal values from the scripts, but not macros or hard-coded values.

I fact I found that named values can also be variable, so in theory, even formulas can be encoded into named values. But then you would needd a context to get the correct value and at least so far the AI cannot specify a context.

Reply:
You'd need a way to specify a context if you want to evaluate a non-constant valueref using something other than a default empty context. The API already does the default empty context case when called for a non-constant value ref. That could already be useful / work for something that can be evaluated with a default empty context, which might be useful for a formula (a combination of math operations) that depends on the current game turn, or other general gamestate info that does not reference `Source` or `Target` objects or a `Value` reference. Hard-coded species, hull, part, special, and similar names could be used, and probably statistics or multi-parameter "complex" variables that just use those hard-coded names, like HullFuel or GameRule lookups in an expression.